Include `Chunk` class source (#38334)
dotnet/docs: docs/csharp/linq/group-results-by-contiguous-keys.md

### `Chunk` class

The `Chunk` class is a contiguous group of one or more source elements that have the same key. A Chunk has a key and a list of ChunkItem objects, which are copies of the elements in the source sequence:

:::code language="csharp" source="../../../samples/snippets/csharp/concepts/linq/LinqSamples/GroupByContiguousKeys.cs" id="group_by_contiguous_keys_chunk_class":::

> A Chunk has a linked list of `ChunkItem`s, which represent the elements in the current chunk. Each `ChunkItem` (represented by `ChunkItem` class) has a reference to the next `ChunkItem` in the list.

Reviewer: ```suggestion
The `ChunkItem` objects are stored in a linked list, where each `ChunkItem` has a reference to the next `ChunkItem` in the list.
```
